include/sycl/algorithm/inclusive_scan.hpp

```diff
@@ -107,7 +109,258 @@ OutputIterator inclusive_scan(ExecutionPolicy &sep, InputIterator b,
   q.wait_and_throw();
   return o + vectorSize;
 }
+#else
 
+
+struct mapscan_descriptor {
+  size_t size, size_per_work_group, size_per_work_item,
+    nb_work_group, nb_work_item;
+  mapscan_descriptor(size_t size_,
+                     size_t size_per_work_group_,
+                     size_t size_per_work_item_,
+                     size_t nb_work_group_,
+                     size_t nb_work_item_):
+    size(size_),
+    size_per_work_group(size_per_work_group_),
+    size_per_work_item(size_per_work_item_),
+    nb_work_group(nb_work_group_),
+    nb_work_item(nb_work_item_) {}
+};
```

Comment:
I like the idea of using descriptors to have all the preparation of sizes and workgroups in a single place, but seems to me having a different descriptor for all functions is an overkill. Is it not possible to have a single descriptor to dominate them all? Seems most of them have a similar interface, so you could maybe have a sycl_algorithm_descriptor structure defined in the ExecutionPolicy with different constructors for the different use cases?